pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -26,10 +36,177 @@ def synchronize(remote_pk, repository_pk):
     """
     remote = RpmRemote.objects.get(pk=remote_pk)
     repository = Repository.objects.get(pk=repository_pk)
-    # base_version = RepositoryVersion.latest(repository)
 
     if not remote.url:
         raise ValueError(_('A remote must have a url specified to synchronize.'))
 
     log.info(_('Synchronizing: repository={r} remote={p}').format(
         r=repository.name, p=remote.name))
+
+    first_stage = RpmFirstStage(remote)
+    DeclarativeVersion(first_stage, repository).create()
+
+
+class RpmFirstStage(Stage):
+    def __init__(self, remote):
+        """
+        The first stage of a pulp_rpm sync pipeline.
+
+        Args:
+            remote (RpmRemote): The remote data to be used when syncing
+
+        """
+        self.remote = remote
+
+    async def __call__(self, in_q, out_q):
```

Comment:
This method contains multiple code blocks each doing something complex.  Recommend decomposing into smaller methods/functions each doing *one* thing.  Proper functional decomposition will be more readable, testable and maintainable.

Reply:
@jortel  This may have been addressed already, this PR cloned and then merged and a few changes have been made on top of that since then.  I think we're leaving this thread open atm because there's still some unadressed discussion left.